atomicapp/nulecule/main.py

```diff
@@ -183,6 +184,11 @@ def run(self, cli_provider, answers_output, ask,
         # it does exist it will be just be loaded and returned
         self.nulecule = self.unpack(dryrun=dryrun, config=self.answers)
 
+        # If no provider is specified within answers.conf or --provider=
+        # we should try and get a default one from the artifacts folder
+        if self.answers[GLOBAL_CONF][PROVIDER_KEY] is None and cli_provider is None:
+            self.answers[GLOBAL_CONF][PROVIDER_KEY] = Utils.getDefaultArtifact(self.app_path)
+
```

Comment:
I believe what you are doing in getDefaultArtifact and what I have above is essentially the same thing because Utils.getAppSupportedProviders would just return the list of providers that there are artifacts for. This would allow us to leave DEFAULT_ANSWERS the way they were in the past and still get the same functionality.

Let's chat about this tomorrow if you like.